mani_skill/envs/tasks/rotate_cube.py

```diff
@@ -0,0 +1,438 @@
+        )
+    )
+
+    # set some commonly used values
+    goal_radius = 0.02
+    cube_half_size = 0.02
+
+    # radius of the area
+    ARENA_RADIUS = 0.195
+    size = 0.065  # m
+    max_len = 0.065
+    # 3D radius of the cuboid
+    radius_3d = max_len * np.sqrt(3) / 2
+    # compute distance from wall to the center
+    max_com_distance_to_center = ARENA_RADIUS - radius_3d
+    # minimum and maximum height for spawning the object
+    min_height = 0.065 / 2
+    max_height = 0.1
+
+    def __init__(
+            self,
+            *args,
+            robot_uids="trifingerpro",
+            robot_init_qpos_noise=0.02,
+            difficulty_level: int = 4,
+            **kwargs
+    ):
+        self.robot_init_qpos_noise = robot_init_qpos_noise
+
+        if (
+                not isinstance(difficulty_level, int)
+                or difficulty_level >= 5
+                or difficulty_level < 0
+        ):
+            raise ValueError(
+                f"Difficulty level must be a int within 0-4, but get {difficulty_level}"
+            )
+
+        self.difficulty_level = difficulty_level
+        super().__init__(*args, robot_uids=robot_uids, **kwargs)
+
+    @property
+    def _sensor_configs(self):
+        pose = look_at(eye=(0.7, 0.0, 0.7), target=(0.0, 0.0, 0.0))
+        return [
+            CameraConfig("base_camera", pose, 128, 128, np.pi / 2, 0.01, 100)
+        ]
+
+    @property
+    def _human_render_camera_configs(self):
+        pose = look_at((0.7, 0.0, 0.7), (0.0, 0.0, 0.0))
+        return CameraConfig("render_camera", pose, 512, 512, 1, 0.01, 100)
+
+    def _load_scene(self, options: dict):
+        loader1 = self._scene.create_urdf_loader()
+        loader1.fix_root_link = True
+        loader1.name = "table"
+        urdf_path = f"{PACKAGE_ASSET_DIR}/robots/trifinger/table_without_border.urdf"
+        table: Articulation = loader1.load(urdf_path)
+
+        builder: ActorBuilder = self._scene.create_actor_builder()
+        high_table_boundary_file_name = f"{PACKAGE_ASSET_DIR}/robots/trifinger/robot_properties_fingers/meshes/high_table_boundary.stl"
+        builder.add_nonconvex_collision_from_file(filename=high_table_boundary_file_name, scale=[1, 1, 1], material=None)
+        builder.add_visual_from_file(filename=high_table_boundary_file_name)
+        table_boundary: Actor = builder.build_static("table2")
+
+        self.obj = actors.build_colorful_cube(
+            self._scene,
+            half_size=self.size / 2,
+            color=np.array([169, 42, 12, 255]) / 255,
+            name="cube",
+            body_type="dynamic",
+            add_collision=True,
+        )
+
+        self.obj_goal = actors.build_colorful_cube(
+            self._scene,
+            half_size=self.size / 2,
+            color=np.array([12, 42, 160, 255]) / 255,
+            name="cube_goal",
+            body_type="kinematic",
+            add_collision=False
+        )
```

Comment:
I recommend making goals a green color (unless you have a good reason not to use green). I think this is the blue color from some of the other tasks we had iirc.

Another thing is because this is actually not part of an image observation, this goal should be added to the hidden objects list, which are all hidden when rendering visual observations (but shown during render_mode=="rgb_array" or "human")

```
self._hidden_objects.append(self.obj_goal)
```